### ISSUE TITLE
Gradle plugin no longer supporting Java 1.8 with 1.17.0-RC1, by accident?

### ISSUE THREAD

**Issue by davidburstrom** [detekt/detekt#3731]
## Expected Behavior

Maybe it is a deliberate change to drop Java 1.8 support, but nothing was mentioned in the release notes about it.

## Observed Behavior

Trying to resolve Gradle plugin 1.17.0-RC1 with JDK 1.8 yields:

```
   > Could not resolve io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.17.0-RC1.
     Required by:
         project :
      > No matching variant of io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.17.0-RC1 was found. The consumer was configured to find a runtime of a library compatible with Java 8, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '7.0' but:
          - Variant 'apiElements' capability io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.17.0-RC1 declares a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares an API of a component compatible with Java 11 and the consumer needed a runtime of a component compatible with Java 8
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.0')
```

## Your Environment
* Version of detekt used: 1.17.0-RC1
* Version of Gradle used (if applicable): 7.0
* Operating System and version: macOS 11.2.3

**Comment by cortinico** [comment]
Yup it was an accident :/ Due to some refactoring we removed the explicit declaration of the Java level. Fix is in #3732 and an RC2 is forthcoming (as soon as #3730 is also shipped).

**Comment by davidburstrom** [comment]
Thanks Nicola! 👋
